File: 01-Assignment-02-02-2026/Using_C/program_30.c
/* Program 30: Check leap year */

#include <stdio.h>
int isLeapYear(int year){
    if ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0){
        return 1;
    }
    return 0;
}

int main() {
    int year;
    printf("Enter Any Valid Year ! : ");
    scanf("%d",&year);

    if (year <= 0){
        printf("Invalid Year !");
        return 0;
    }
    if (isLeapYear(year)){
        printf("%d is Leap Year.",year);
    }
    else{
        printf("%d is not Leap Year.",year);
    }
    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program30.java
/*
Program 30
Java program to check leap year
*/

import java.util.Scanner;

public class Program30 {

    public static boolean isLeap(int year) {
        return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a valid year: ");
        int year = scanner.nextInt();

        if (year <= 0) {
            System.out.println("Invalid year");
        } else {
            if (isLeap(year)) {
                System.out.println(year + " is a leap year");
            } else {
                System.out.println(year + " is not a leap year");
            }
        }

        scanner.close();
    }
}